### PR TITLE
Breadcrumbs refactor for register an operation

### PR DESCRIPTION
**Pull request by shon-button**
Addresses: 
[2063](https://github.com/bcgov/cas-registration/issues/2063)

### 🚀 Impact:
- Refactored  `Bread` for `register-an-operation`
- Added `Bread` vitest

### 🔬 Local Testing:

1. From terminal command, start the api server:
 ```
cd bc_obps
make reset_db
make run
```  
2.  From terminal command, start the app development server:
 ```
cd client && yarn dev-all
```  
 
### Test Url Links:

- http://localhost:3000/registration/register-an-operation
image

- http://localhost:3000/registration/register-an-operation/e8f913c5-7636-42f8-92f1-b90cad5f9ecc/2
image

- http://localhost:3000/administration/contacts/10?contacts_title=Henry%20Ives
image

- http://localhost:3000/administration/contacts/add-contact
image

- http://localhost:3000/administration/operations
image

- http://localhost:3000/administration/operations/002d5a9e-32a6-4191-938c-2c02bfec592d/facilities?operations_title=Operation+2
image

- http://localhost:3000/administration/operations/002d5a9e-32a6-4191-938c-2c02bfec592d/facilities/f486f2fb-62ed-438d-bb3e-0819b51e3aeb?operations_title=Operation%202&facilities_title=Facility%201
image

### CODE REVIEW

File: bciers/libs/components/src/navigation/Bread.tsx [hunk 1]
```diff
@@ -67,6 +41,20 @@ export default function Bread({
   const searchParams = useSearchParams();
   const crumbTitles = Object.fromEntries(searchParams.entries());
 
+  // 🛠️ Function to check if a link is the last link
+  const isLastBreadcrumbItem = (link: string, index: number) => {
+    const lastLinkValues = ["register-an-operation"];
```
Comment (marcellmueller) [inline bciers/libs/components/src/navigation/Bread.tsx:46]: For cleanliness sake should we make this an argument that we pass in the `layout` similar to `defaultLinks`?

**Comment by marcellmueller** [comment]
Looks great, nice work! 

I think it's safe to change the `/registration/operation` route to `/registration/register-an-operation` in this PR. Shouldn't cause any issues with in progress PRs.

**Comment by shon-button** [comment]
> Looks great, nice work!
> 
> I think it's safe to change the `/registration/operation` route to `/registration/register-an-operation` in this PR. Shouldn't cause any issues with in progress PRs.

@marcellmueller 

I always appreciate your thoughts; although, could we differ the route change in the PR and let Bri's PR handle it?

...it's been a long CI test day and I fear running the CI pipeline again  😰

**Comment by marcellmueller** [comment]
> > Looks great, nice work!
> > I think it's safe to change the `/registration/operation` route to `/registration/register-an-operation` in this PR. Shouldn't cause any issues with in progress PRs.
> 
> @marcellmueller
> 
> I always appreciate your thoughts; although, could we differ the route change and let Bri's PR handle it?
> 
> ...it's been a long CI test day and I fear running the CI pipeline again 😰

Sounds good. I still think we could consider passing `lastLinkValues` in the layout like `defaultLinks`, but I also have empathy for CI issues 😆 

Excellent work! 🥇

**Review by marcellmueller — APPROVED** [review]

File: bciers/libs/components/src/navigation/Bread.tsx (hunk 1 above)
Comment (shon-button) [reply bciers/libs/components/src/navigation/Bread.tsx:46]: @marcellmueller 

A good thought...
Although, I see it as more of a static list...but, of course, I am biased by a wee anti props drilling attitude.

**Pull request merged**